test/jdk/java/lang/Math/WorstCaseTests.java

```diff
@@ -243,6 +261,9 @@ private static int testWorstCos() {
             {+0x1.7CB7648526F99p-1,     +0x1.78DAF01036D0Cp-1},
             {+0x1.C65A170474549p-1,     +0x1.434A3645BE208p-1},
             {+0x1.6B8A6273D7C21p+0,     +0x1.337FC5B072C52p-3},
+
+            // Worst-case observed error
+            {-0x1.34e729fd08086p+21,    +0x1.6a6a0d6a17f0fp-1},
```

Comment:
Both `Math.cos` and `StrictMath.cos` produce the correctly rounded result here.
I don't know why the paper says otherwise. Perhaps OpenLibm is not exactly fdlibm.

Reply:
> Both `Math.cos` and `StrictMath.cos` produce the correctly rounded result here. I don't know why the paper says otherwise. Perhaps OpenLibm is not exactly fdlibm.

I've looked a bit over the OpenLibm changelog. They've added a few special cases for exp and pow as leaset. OpenLibm itself is a direct derivative of one of the BSD math libraries, which in turn was derived from FDLIBM.

Reply:
> > Both `Math.cos` and `StrictMath.cos` produce the correctly rounded result here. I don't know why the paper says otherwise. Perhaps OpenLibm is not exactly fdlibm.
> 
> I've looked a bit over the OpenLibm changelog. They've added a few special cases for exp and pow as least. OpenLibm itself is a direct derivative of one of the BSD math libraries, which in turn was derived from FDLIBM.

The BSDs look to have changed/improved the kernel cos computation in their fork.